src/libraries/Common/src/Interop/Interop.Zstd.cs

```diff
@@ -0,0 +1,383 @@
+
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial nuint ZSTD_DCtx_setParameter(SafeZstdDecompressHandle dctx, ZstdDParameter param, int value);
+
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial nuint ZSTD_DCtx_refPrefix(SafeZstdDecompressHandle dctx, IntPtr prefix, nuint prefixSize);
+
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial nuint ZSTD_DCtx_reset(SafeZstdDecompressHandle dctx, ZstdResetDirective reset);
+
+        // Streaming compression
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial nuint ZSTD_compressStream2(SafeZstdCompressHandle cctx, ref ZstdOutBuffer output, ref ZstdInBuffer input, ZstdEndDirective endOp);
+
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial nuint ZSTD_CCtx_setParameter(SafeZstdCompressHandle cctx, ZstdCParameter param, int value);
+
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial nuint ZSTD_CCtx_refPrefix(SafeZstdCompressHandle cctx, IntPtr prefix, nuint prefixSize);
+
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial nuint ZSTD_CCtx_reset(SafeZstdCompressHandle cctx, ZstdResetDirective reset);
+
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial nuint ZSTD_CCtx_setPledgedSrcSize(SafeZstdCompressHandle cctx, nuint pledgedSrcSize);
+
+        // Compression level functions
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial int ZSTD_minCLevel();
+
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial int ZSTD_maxCLevel();
+
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial int ZSTD_defaultCLevel();
+
+        // Error checking
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial uint ZSTD_isError(nuint result);
+
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial IntPtr ZSTD_getErrorName(nuint result);
+
+        // Dictionary context functions
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial nuint ZSTD_DCtx_refDDict(SafeZstdDecompressHandle dctx, SafeZstdDDictHandle ddict);
+
+        [LibraryImport(Libraries.CompressionNative)]
+        internal static partial nuint ZSTD_CCtx_refCDict(SafeZstdCompressHandle cctx, SafeZstdCDictHandle cdict);
+
+        // Enums and structures for streaming
+        internal enum ZstdEndDirective
+        {
+            ZSTD_e_continue = 0,
+            ZSTD_e_flush = 1,
+            ZSTD_e_end = 2
+        }
+
+        internal enum ZstdCParameter
+        {
+            ZSTD_c_compressionLevel = 100, /* Set compression parameters according to pre-defined cLevel table.
```

Comment:
Why not make these /// to benefit internal usage?

Reply:
Done